### ISSUE TITLE
coreos: fleet: obsolete version(s) included by default, thus omitting fleet bug fixes

### ISSUE THREAD

**Issue by megahall** [coreos/bugs#1448]
# Description

A Fleet bug fix which affects use of `fleetctl destroy` in my deployment is available upstream in `Fleet v0.12.0` and up, but the CoreOS `Fleet` is quite far behind the upstream Fleet.

Would it be possible either to backport the fix in coreos/fleet#1417 into `Fleet v0.11.x` or ship a standard train CoreOS which includes `Fleet 0.12.x` or up? Because no current version of CoreOS includes `Fleet 0.12.x` or up as of yet.

https://github.com/coreos/fleet/issues/1383#issuecomment-230618528
https://github.com/coreos/fleet/pull/1417#issuecomment-230618567
## CoreOS Version

Issue applies to all published CoreOS releases up to and including the current latest, `1097.0.0`. This can be confirmed by verifying the `Fleet` versions listed in all releases noted in the release notes at https://coreos.com/releases/ .
## Environment

This is a base dependency issue which is independent of hardware, cloud, or hypervisor releases.
## Expected Behavior

CoreOS should include more recent branches of upstream `Fleet` on a regular basis to prevent missing important bug fixes.
## Actual Behavior

CoreOS does not include more recent branches of upstream `Fleet` on a regular basis to prevent missing important bug fixes.
## Reproduction Steps

_NOTE_: See https://github.com/coreos/fleet/issues/1383 for more details.
1. Run `fleetctl destroy` on a service that's not active for whatever reason.
2. Obsolete copies of `Fleet` print storms of `Error destroying units: googleapi: Error 404: unit does not exist` messages and return spurious POSIX process return codes.
3. Current copies of `Fleet` do not print the messages or return errors and just work.
## Other Information

N/A.

**Comment by crawford** [comment]
We aren't going to update the version of fleet in Container Linux anymore. We are going to sunset the project and remove it from the OS.

**Comment by codepunkt** [comment]
@crawford What does this mean exactly? We're setting up a CoreOS cluster at the moment. Should we continue using Fleet or rather switch to Kubernetes or some other solution? Is there a timeframe for this or an official announcement? What is the reasoning behind it?

**Comment by crawford** [comment]
Early on, we built fleet to help bridge the gap between single-instance scheduling (solved by systemd) and cluster-wide scheduling. Since then, kubernetes was released and it was clear to us that this was the superior technology and this is what we would focus our efforts on instead. I've been fighting to officially sunset the fleet project for a long time now and we finally have a roadmap. Unfortunately, it has taken far too long to relay that plan to our users and I feel forced to warn people now.

All new projects should not use fleet. Our official recommendation is kubernetes. It can do everything fleet can do (and much more) except it can do it much quicker and more efficiently.

We will be making an official announcement early next year and will provide details on the timeline, how to contribute using fleet, and how to migrate to kubernetes.
